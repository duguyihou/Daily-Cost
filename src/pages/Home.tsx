import { FeatOne } from '@features/featOne'
import React from 'react'
import { View } from 'react-native'

function Home() {
  return (
    <View>
      <FeatOne />
    </View>
  )
}
export default Home
