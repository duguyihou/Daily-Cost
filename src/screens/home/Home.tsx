import { Bill } from '@features/bill'
import { YearCardList } from '@features/year'
import { useNavigation } from '@react-navigation/native'
import { HomeStackNavigationProps, RouteName } from '@routes/Routes.types'
import RealmContext from '@shared/RealmContext'
import dayjs from 'dayjs'
import React, { useEffect } from 'react'
import { StyleSheet, View } from 'react-native'

function Home() {
  const navigation = useNavigation<HomeStackNavigationProps>()
  useEffect(() => {
    const year = dayjs().format('YYYY')
    navigation.push(RouteName.Month, { year, ignore: false })
  })
  const { useQuery } = RealmContext
  const bills = useQuery(Bill)
  console.log('üêµ bills ------ ', bills)
  return (
    <View style={styles.container}>
      <YearCardList />
    </View>
  )
}
export default Home

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
})
