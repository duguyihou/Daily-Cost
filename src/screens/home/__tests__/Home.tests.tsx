import { render } from '@testing-library/react-native'
import React from 'react'

import Home from '../Home'

jest.mock('@features/bill', () => ({
  Card: jest.fn(),
}))
test('home', () => {
  const home = render(<Home />)
  expect(home).toMatchSnapshot()
})
