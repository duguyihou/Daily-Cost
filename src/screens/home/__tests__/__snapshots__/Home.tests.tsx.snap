// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "left",
          }
        }
      >
        January 2020
      </Text>
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "textAlign": "left",
          }
        }
      >
        109.90
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Coles
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      29.90
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      WWS
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      29.90
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 20,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      KFC
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      13.90
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ef928f",
        "borderRadius": 30,
        "bottom": 30,
        "opacity": 1,
        "padding": 10,
        "position": "absolute",
        "right": 30,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="28"
      bbWidth="28"
      focusable={false}
      height={28}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 28,
            "width": 28,
          },
        ]
      }
      vbHeight={512}
      vbWidth={448}
      width={28}
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
          fill={
            Object {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
